src/mesh/distributed_mesh.C

```diff
@@ -189,6 +192,7 @@ DistributedMesh::DistributedMesh (const DistributedMesh & other_mesh) :
 DistributedMesh::DistributedMesh (const UnstructuredMesh & other_mesh) :
   UnstructuredMesh (other_mesh), _is_serial(other_mesh.is_serial()),
   _is_serial_on_proc_0(other_mesh.is_serial()),
+  _deleted_coarse_elements(true), // better safe than sorry...
```

Comment:
Hmm... in a copy from a `ReplicatedMesh`, the original wouldn't have this problem, because it would naturally see all elements connected to a node and would never try to orphan anything that had even a weird non-local connection ... but in a copy of the exact same elements?  I think the subsequent remote element deletion would just turn `_delete_coarse_elements` back on, so we wouldn't have a bug, but we also wouldn't end up taking advantage of skipping that sync step.

... and now that I think about it, that's actually a performance bug in other contexts.  `delete_remote_elements` is doing its `delete_elem` calls carefully, in a way that isn't going to trigger the problem that that sync step is necessary to repair.  I need to only be turning on `_deleted_coarse_elements` when a *local* element is deleted!

I'll add a commit to do that, and verify that it still fixes the MOOSE problem.

Reply:
Still fixes the MOOSE problem!